src/lib/nvme_pt_ioctl.c

```diff
@@ -414,9 +413,18 @@ int opal_init_pt(struct sed_device *dev, const char *device_path)
 	return ret;
 }
 
-void opal_level0_discv_info_pt(struct sed_opal_level0_discovery *discvry)
+int opal_level0_discv_info_pt(struct sed_device *dev,
+				struct sed_opal_level0_discovery *discv)
 {
-	memcpy(discvry, (struct sed_opal_level0_discovery *)discv, sizeof(*discvry));
+	int ret;
+
+	ret = opal_level0_disc_pt(dev);
+	if (ret)
+		return ret;
+
+	memcpy(discv, &dev->discv, sizeof(*discv));
```

Comment:
before doing memcpy you could check if if input is not null

Reply:
sorry for missing out this check condition